integration_tests/suite/base/test_users.py

```diff
@@ -2417,3 +2417,59 @@ def test_update_incall_new_incall():
         ),
     )
     url.delete(recursive=True)
+
+
+@fixtures.device()
+@fixtures.device()
+def test_post_lines_same_unallocated_device_no_error(device, device2):
+    user_resources = generate_user_resources_bodies(context_name=CONTEXT, device=device)
+
+    # define another line with the same extension
+    line2 = {
+        'context': CONTEXT,
+        'extensions': [user_resources.extension],
+        'endpoint_sip': {'name': s.random_string(5)},
+    }
+    line2['device_id'] = device2['id']
+
+    # user creation with 2 lines
+    response = confd.users.post(
+        {
+            'lines': [user_resources.line, line2],
+            **user_resources.user,
+        }
+    )
+
+    response.assert_created('users')
+    payload = response.item
+
+    # check if the returned data contains the device_id
+    assert_that(
+        payload,
+        has_entries(
+            uuid=uuid_(),
+            lines=contains_inanyorder(
+                has_entries(
+                    device_id=device['id'],
+                ),
+                has_entries(
+                    device_id=device2['id'],
+                ),
+            ),
+            **user_resources.user,
+        ),
+    )
+
+    # retrieve the lines (created before) and check if the device is associated to the line
+    assert_that(
+        confd.lines(payload['lines'][0]['id']).get().item,
+        has_entries(device_id=device['id']),
+    )
+    assert_that(
+        confd.lines(payload['lines'][1]['id']).get().item,
+        has_entries(device_id=device2['id']),
+    )
```

Comment:
Prefer code styling instead of comment. ex:
```suggestion
    existing_line_1 = confd.lines(payload['lines'][0]['id']).get().item
    added_device_id_1 = device['id']
    assert_that(existing_line_1, has_entries(device_id=added_device_id_1))
    
    existing_line_2 = confd.lines(payload['lines'][1]['id']).get().item
    added_device_id_2 = device2['id']
    assert_that(existing_line_2, has_entries(device_id=added_device_id_2))
```
**Note: feel free to change name - not tested
**Note: Refering to previous comment, I'm not sure we should check device_id instead of extension

Reply:
assert on extension instead of device and comment removed